### DISCUSSION TITLE
curl performs only a limited number of requests on files

### DISCUSSION THREAD

**Discussion by tombstoneghost** [curl/curl#6415] [Q&A] [answered]
Is it that the curl will only perform 255 Requests if we are trying to check the availability of various web pages. 

Requests like: http://127.0.0.1/hello.txt

We can perform only 255 such requests?

**Comment by bagder** [comment, marked as answer]
No. curl will do virtually unlimited amounts of requests if you just tell it to and the server is okay with it.

**Reply by tombstoneghost** [reply]
But I get this error after 255 requests. 

ERROR: Couldn't connect to server       [0]

Here, [0] is the response code and there is no such response code as 0.

**Reply by bagder** [reply]
As the error message says, curl failed to connect to the server. It would indicate that the server doesn't like more connections done to it within a certain time interval or perhaps from that same IP or something. It is not a limitation in curl.
